### PR TITLE
Refactor get_map_coordinates_from_pos_window

### PR DESCRIPTION
**Pull request by bentglasstube**
This changes get_map_coordinates_from_pos_window to use the
ScreenCoordsXY and CoordsXY structs instead of bare ints as suggested
in #10065

### CODE REVIEW

File: src/openrct2/interface/Viewport.cpp [hunk 1]
```diff
@@ -1665,25 +1665,40 @@ void get_map_coordinates_from_pos(
 void get_map_coordinates_from_pos_window(
     rct_window* window, int32_t screenX, int32_t screenY, int32_t flags, int16_t* x, int16_t* y, int32_t* interactionType,
     TileElement** tileElement, rct_viewport** viewport)
+{
+  ScreenCoordsXY screenCoords(screenX, screenY);
+  CoordsXY mapCoords(0, 0);
```
Comment (tupaschoal) [inline src/openrct2/interface/Viewport.cpp]: `CoordsXY` are already initialized to {0, 0} on their default constructor, so this seems superfluous

Comment (bentglasstube) [reply src/openrct2/interface/Viewport.cpp]: Good point, changed to use the default constructor.

File: src/openrct2/interface/Viewport.cpp [hunk 2]
```diff
@@ -1665,25 +1665,40 @@ void get_map_coordinates_from_pos(
 void get_map_coordinates_from_pos_window(
     rct_window* window, int32_t screenX, int32_t screenY, int32_t flags, int16_t* x, int16_t* y, int32_t* interactionType,
     TileElement** tileElement, rct_viewport** viewport)
+{
+  ScreenCoordsXY screenCoords(screenX, screenY);
+  CoordsXY mapCoords();
```
Comment (duncanspumpkin) [inline src/openrct2/interface/Viewport.cpp]: `{}` or just don't put anything. At present it is trying to interpret this as a function definition.

**Comment by duncanspumpkin** [comment]
just need to clang format the changes.

**Comment by duncanspumpkin** [comment]
argh. You've got the same version of clang format as I have. It isn't the same version as what is installed on the CI. You will have to manually fix the file yourself.

**Comment by Gymnasiast** [comment]
Could you rebase this?

**Comment by duncanspumpkin** [comment]
I've fixed the conflict and will squash merge to remove the merge commit.

**Comment by bentglasstube** [comment]
oh thanks, I didn't have a chance to rebase it.

--
Alan Berndt

**Pull request merged**